File: src/Starfield2026.BgEditor/backend/src/lib/ohana/Models/PocketMonsters/CP.ts
/**
 * Loads a CP overworld character model from Pokemon.
 * Ported from OhanaCli.Formats.Models.PocketMonsters.CP (C#).
 */

import { BinaryReader } from '../../Core/BinaryReader.js';
import { PkmnContainer } from '../../Containers/PkmnContainer.js';
import { CM } from './CM.js';
import { OModelGroup } from '../../Core/RenderBase.js';

export class CP {
  static load(data: BinaryReader): OModelGroup {
    const container = PkmnContainer.load(data);
    return CM.load(BinaryReader.fromBuffer(container.content[1].data!));
  }
}

File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Garc/Models/PocketMonsters/CP.cs
using MiniToolbox.Garc.Containers;

using System.IO;

namespace MiniToolbox.Garc.Models.PocketMonsters
{
    class CP
    {
        /// <summary>
        ///     Loads a CP overworld character model from Pokémon.
        /// </summary>
        /// <param name="data">The data</param>
        /// <returns>The Model group with the character meshes</returns>
        public static RenderBase.OModelGroup load(Stream data)
        {
            RenderBase.OModelGroup models = new RenderBase.OModelGroup();

            OContainer container = PkmnContainer.load(data);
            models = CM.load(new MemoryStream(container.content[1].data));

            return models;
        }
    }
}
